src/poliastro/plotting/porkchop.py

```diff
@@ -17,12 +18,23 @@ def _targetting(departure_body, target_body, t_launch, t_arrival):
     """
 
     # Compute departure and arrival positions
-    rr_dpt_body, vv_dpt_body = coord.get_body_barycentric_posvel(
-        departure_body.name, t_launch
-    )
-    rr_arr_body, vv_arr_body = coord.get_body_barycentric_posvel(
-        target_body.name, t_arrival
-    )
+    if isinstance(departure_body, Orbit):
```

Comment:
Rather than all these `isinstance` checks, which are extremely slow and complicate the code, I would create a new function for the new interface. And if there's lots of duplicate code between the new and the current one, then there's an opportunity to refactor and put the common code in a private function, for example.

Reply:
Sorry for the delay :sweat_smile:  I finally wrote a private function called `_get_state` that returns the state vectors (rr, vv) of a body depending if it belongs to `Orbit` or `Body` instances. This function does not make use of `isinstance` but requires importing all poliastro bodies instead.

Reply:
Fair enough! If this piece of code is going away, please remove it (don't leave it as a multistring comment)